File: 20180224-mars-rover-kata/src/test/java/com/katas/MarsRoverTest.java
package com.katas;

import org.junit.Before;
import org.junit.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class MarsRoverTest {

    MarsRover rover;

    @Before
    public void setUp() {
        rover = new MarsRover();
    }

    @Test
    public void should_indicate_current_position() {
        GridPosition position = rover.currentPosition();

        assertThat(position).isEqualTo(new GridPosition(0, 0));
    }

    @Test
    public void should_indicate_current_facing_direction() {
        String direction = rover.currentDirection();

        assertThat(direction).isEqualTo("N");
    }

    @Test
    public void should_move_forward() {
        rover.handleCommand(new char[]{'f'});

        assertThat(rover.currentPosition()).isEqualTo(new GridPosition(0, 1));
    }

    @Test
    public void should_move_forward_multiple_steps() {
        rover.handleCommand(new char[]{'f', 'f', 'f'});

        assertThat(rover.currentPosition()).isEqualTo(new GridPosition(0, 3));
    }

    @Test
    public void should_move_backward() {
        rover.handleCommand(new char[]{'b'});

        assertThat(rover.currentPosition()).isEqualTo(new GridPosition(0, -1));
    }

}

File: 20220921-marsrover-kotlin/src/test/kotlin/com/katas/MarsRoverTest.kt
package com.katas

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

class MarsRoverTest {

    @Test
    fun `should move forward one square`(){
        val rover = MarsRover(Position(0, 0))

        rover.receiveCommand("f")

        assertThat(rover.position).isEqualTo(Position(0 ,1))
    }

    @Test
    fun `should move forward multiple square`(){
        val rover = MarsRover(Position(0, 100))

        rover.receiveCommand("ff")

        assertThat(rover.position).isEqualTo(Position(0 ,102))
    }

    @Test
    fun `should move backward one square`(){
        val rover = MarsRover(Position(0, 0))

        rover.receiveCommand("b")

        assertThat(rover.position).isEqualTo(Position(0 ,-1))
    }
}
